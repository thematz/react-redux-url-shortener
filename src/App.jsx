import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { fetchUrl } from './actions'
import Form from './components/Form'
import List from './components/List'
import Notification from './components/Notification'
import 'bulma/css/bulma.css'

class App extends Component {
	componentWillMount() {
		if (this.props.location.pathname !== '/') {
			this.props.fetchUrl(this.props.location.pathname)
		}
	}
	render() {
		const { notification } = this.props
		return (
			<div className="App">
				<main>
					<Notification type={notification.type} message={notification.message} />
					<Form />
					<List />
				</main>
			</div>
		)
	}
}

const mapStateToProps = state => ({
	notification: state.notification
})

const mapDispatchToProps = dispatch => bindActionCreators({
	fetchUrl
}, dispatch)

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(App)