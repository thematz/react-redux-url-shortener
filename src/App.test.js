import React from 'react';
import App from './App';
import store from './store'

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() })

it('renders without crashing', () => {
    const wrapper = shallow(
        <App store={store} />,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
});
