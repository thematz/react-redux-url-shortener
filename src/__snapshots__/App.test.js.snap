// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<App
  fetchUrl={[Function]}
  notification={Object {}}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
